### PR TITLE
Step_5: added brain-even games

### PR DESCRIPTION
**Pull request by alekseifeniuk**
Hi, good evening!

https://ru.hexlet.io/projects/49/members/11286?step=5

### CODE REVIEW

File: brain_games/cli.py [hunk 1]
```diff
@@ -1,9 +1,41 @@
...
+    user_name = prompt.string('May I have your name? ')
+    print(f'Hello, {user_name}!')
+    return user_name
+
+
+def greet_user():
+    print(f'Congratulations, {user_name}!')
+
+
+def is_even():
```
Comment (svntmr) [inline brain_games/cli.py:21]: > cli.py не предназначен для описания логики игр.    

consider moving this function into some file in a special folder for games backend/logic, e. g. `brain_games/games`

File: brain_games/cli.py [hunk 2]
```diff
@@ -1,9 +1,41 @@
...
 import prompt
 
+user_name = ''
+
 
 def welcome_user():
     print("Welcome to the Brain Games!")
-    name = prompt.string('May I have your name? ')
-    print('Hello, {}!'.format(name))
+    global user_name
```
Comment (svntmr) [inline brain_games/cli.py:11]: global variables is a bad practice, consider refactoring welcome_user to return `user_name`. `user_name` definition from line 6 should be removed

File: brain_games/cli.py [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+    user_name = prompt.string('May I have your name? ')
+    print(f'Hello, {user_name}!')
+    return user_name
+
+
+def greet_user():
```
Comment (svntmr) [inline brain_games/cli.py:17]: this function also should be refactored as `user_name` won't be global. do smth to pass user name into this function

Comment (svntmr) [reply brain_games/cli.py:17]: I'd rename it to `congratulate_user`

File: brain_games/cli.py [hunk 4]
```diff
@@ -1,9 +1,41 @@
...
+    for i in range(3):
+        random_number = randint(1, 500)
+        print(f'Question: {random_number}')
+
+        if random_number % 2 == 0:
+            right_answer = 'yes'
+        else:
+            right_answer = 'no'
+
+        user_answer = prompt.string('You answer: ')
```
Comment (svntmr) [inline brain_games/cli.py:32]: You**r**

File: brain_games/cli.py [hunk 5]
```diff
@@ -1,9 +1,41 @@
...
+
+        if user_answer == right_answer:
+            print('Correct!')
+        else:
+            print(f'"{user_answer}" is wrong answer ;(. Correct answer was "{right_answer}".')
+            print(f'Let\'s try again, {user_name}!')
```
Comment (svntmr) [inline brain_games/cli.py:38]: you can use double quotes instead of `\'`

File: pyproject.toml [hunk 6]
```diff
@@ -9,6 +9,7 @@ packages = [
```
Comment (svntmr) [inline pyproject.toml:9]: don't forget to update the project version with each step, `poerty version minor`

Comment (svntmr) [reply pyproject.toml:9]: https://python-poetry.org/docs/cli/#version

**Review by svntmr — CHANGES_REQUESTED** [review]
please, fix the lint action (it fails).   
also, your README is missing asciinema with the game showcase

File: brain_games/games/brain_even_rules.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#!usr/bin/env python3
+
+from random import randint
+
+game_rules = 'Answer "yes" if the number is even, otherwise answer "no".'
+
+
+def rules():
+    game_task = randint(1, 500)
```
Comment (svntmr) [inline brain_games/games/brain_even_rules.py:9]: just to add, it's not necessary to name variables inside the function in the way they're named in caller code (e. g. you can name it `random number` if you want)

File: brain_games/game_core.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
+#!usr/bin/env python3
+
+import prompt
+
+
+def game_core(game):
+    print("Welcome to the Brain Games!")
+    name = prompt.string('May I have your name? ')
+    print(f'Hello, {name}!')
```
Comment (svntmr) [inline brain_games/game_core.py:9]: you can use `welcome_user` from cli module there, just update it to return user name

File: brain_games/game_core.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    print(game.game_rules)
+    for i in range(3):
+        game_task, right_answer = game.rules()
+        print(f'Question: {game_task}')
```
Comment (svntmr) [inline brain_games/game_core.py:13]: I'd move all printing functions into cli module

File: brain_games/game_core.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+        user_answer = prompt.string('Your answer: ')
```
Comment (svntmr) [inline brain_games/game_core.py:14]: prompt stuff should be in cli module as well

File: brain_games/game_core.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+        if user_answer == right_answer:
+            print('Correct!')
```
Comment (svntmr) [inline brain_games/game_core.py:17]: I'd move all printing functions into cli module

File: brain_games/game_core.py [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+        else:
+            a = f'"{user_answer}" is wrong answer ;(. '
+            b = f'Correct answer was "{right_answer}".'
+            print(a + b)
+            print(f"Let's try again, {name}!")
```
Comment (svntmr) [inline brain_games/game_core.py:22]: I'd move all printing functions into cli module

File: brain_games/game_core.py [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+            break
+    else:
+        print(f'Congratulations, {name}!')
```
Comment (svntmr) [inline brain_games/game_core.py:25]: I'd move all printing functions into cli module

**Review by svntmr — CHANGES_REQUESTED** [review]
the structure of the project is much better now!

File: brain_games/cli.py [hunk 14]
```diff
@@ -6,4 +6,32 @@
 def welcome_user():
     print("Welcome to the Brain Games!")
     name = prompt.string('May I have your name? ')
-    print('Hello, {}!'.format(name))
+    print(f'Hello, {name}!')
+    return name
+
+
+def print_rules(game):
```
Comment (svntmr) [inline brain_games/cli.py:13]: I think it's better to pass rules to this function instead of game module

File: brain_games/cli.py [hunk 15]
```diff
@@ -6,4 +6,32 @@
...
+    print(game.game_rules)
+
+
+def question(game_task):
```
Comment (svntmr) [inline brain_games/cli.py:17]: I'd rename it to `ask_question`

File: brain_games/cli.py [hunk 16]
```diff
@@ -6,4 +6,32 @@
...
+    print(f'Question: {game_task}')
+
+
+def answer():
```
Comment (svntmr) [inline brain_games/cli.py:21]: I'd rename it to `ask_answer`

File: brain_games/game_core.py [hunk 17]
```diff
@@ -0,0 +1,23 @@
+#!usr/bin/env python3
+
+from brain_games.cli import welcome_user, print_rules
```
Comment (svntmr) [inline brain_games/game_core.py:3]: you can do multiline imports, like
```python
from ... import (
    a,
    b,
    c
)
```

File: brain_games/games/brain_even_rules.py [hunk 18]
```diff
@@ -0,0 +1,14 @@
+#!usr/bin/env python3
+
+from random import randint
+
+game_rules = 'Answer "yes" if the number is even, otherwise answer "no".'
+
+
+def rules():
```
Comment (svntmr) [inline brain_games/games/brain_even_rules.py:8]: I'd rename it to `get_game_round`

File: brain_games/game_core.py [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+from brain_games.cli import question, answer, correct_answer
+from brain_games.cli import game_over, congratulate_user
+from brain_games.games import brain_even_rules
```
Comment (svntmr) [inline brain_games/game_core.py:6]: I'd rename `brain_even_rules` to `brain_even` as you're importing a game, not its rules

File: brain_games/cli.py [hunk 20]
```diff
@@ -6,4 +6,32 @@
 def welcome_user():
```
Comment (svntmr) [inline brain_games/cli.py:6]: I'd rename it to `welcome_and_acknowledge_user` as it returns user name now

**Review by svntmr — CHANGES_REQUESTED** [review]
looks better and better!

File: brain_games/game_core.py (hunk 19 above)
Comment (alekseifeniuk) [reply brain_games/game_core.py:6]: Hmm, then script file and game file will have similar name. I rename to brain_even_game, alright?

File: README.md [hunk 21]
```diff
@@ -3,3 +3,7 @@
 [![Actions Status](https://github.com/alekseifeniuk/python-project-lvl1/workflows/hexlet-check/badge.svg)](https://github.com/alekseifeniuk/python-project-lvl1/actions)
 [![Maintainability](https://api.codeclimate.com/v1/badges/a99a88d28ad37a79dbf6/maintainability)](https://codeclimate.com/github/alekseifeniuk/python-project-lvl1/maintainability)
 [![Linter Status](https://github.com/alekseifeniuk/python-project-lvl1/actions/workflows/lint.yml/badge.svg)](https://github.com/alekseifeniuk/python-project-lvl1/actions)
+
+### Asciinema brain_even:
+
+https://asciinema.org/a/RjCfq3HtaGA7mBVgeWHPgTpUY
```
Comment (svntmr) [inline README.md:9]: I'd add a new section for games and list games under it. Also, asciinema should be used as a markdown link (see the suggestion)
```suggestion
## Games in the project

### Brain Even
[![asciicast](https://asciinema.org/a/RjCfq3HtaGA7mBVgeWHPgTpUY.svg)](https://asciinema.org/a/RjCfq3HtaGA7mBVgeWHPgTpUY)
```

File: brain_games/game_core.py [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+    ask_question,
+    ask_answer,
+    correct_answer,
+    congratulate_user,
+    game_over)
+
+
+def game_core(game):
+    user_name = welcome_and_acknowledge_user()
+    print_rules(brain_even_game.game_rules)
```
Comment (svntmr) [inline brain_games/game_core.py:16]: this file should be able to work with any game, but now it uses `brain_even_game` rule even if another game will be loaded. 
```suggestion
    print_rules(game.game_rules)
```

File: brain_games/game_core.py [hunk 23]
```diff
@@ -0,0 +1,28 @@
+#!usr/bin/env python3
+
+from brain_games.games import brain_even_game
```
Comment (svntmr) [inline brain_games/game_core.py:3]: this import is not needed

File: brain_games/game_core.py [hunk 24]
```diff
@@ -0,0 +1,28 @@
...
+    ask_answer,
+    correct_answer,
+    congratulate_user,
+    game_over)
+
+
+def game_core(game):
+    user_name = welcome_and_acknowledge_user()
+    print_rules(brain_even_game.game_rules)
+    for i in range(3):
```
Comment (svntmr) [inline brain_games/game_core.py:17]: [optional] python has a special variable for smth not needed - `_`. In this particular case `i` is not needed, you can replace it with `_` if you want. It's not necessary, but I'm looking for `i` usages when I'm seeing the loop with it

**Review by svntmr — CHANGES_REQUESTED** [review]
imho, this is the last review of this PR - you're close to the finish of step 5 :^)

**Comment by svntmr** [comment]
Looks good, ready to be merged!

**Review by svntmr — APPROVED** [review]

**Pull request merged**